### ISSUE TITLE
ltfsck - Cannot open device

### ISSUE THREAD

**Issue by alexhabgood** [LinearTapeFileSystem/ltfs#312]
Hiya,

This is my first time I've attempted to use ltfsck, so this could well be a syntax or compile issue. OS is Ubuntu 18.04 LTS, tape drive is Quantum LTO6 Model C. When I enter `sudo ltfsck /dev/sg7`. It returns an error (see log below). /dev/sg7 is the correct address for the tape drive, since I'm able to format and mount tapes 
with this.

Any advice would be greatly appreciated. Thanks in advance.

```
maceadmin@amacewrk05:/$ sudo ltfsck /dev/sg7
LTFS16000I Starting ltfsck, LTFS version 2.5.0.0 (Prelim), log level 2.
LTFS16088I Launched by "ltfsck /dev/sg7".
LTFS16089I This binary is built for Linux (x86_64).
LTFS16090I GCC version is 7.5.0.
LTFS17087I Kernel version: Linux version 5.4.0-70-generic (buildd@lgw01-amd64-039) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #78~18.04.1-Ubuntu SMP Sat Mar 20 14:10:07 UTC 2021 i386.
LTFS17089I Distribution: NAME="Ubuntu".
LTFS17089I Distribution: DISTRIB_ID=Ubuntu.
LTFS17085I Plugin: Loading "sg" tape backend.
LTFS30209I Opening a device through sg-ibmtape driver (/dev/sg7).
LTFS30288I Opening a tape device for drive serial /dev/sg7.
LTFS12012E Cannot open device: failed backend open call.
LTFS16011E Cannot open device '/dev/sg7'.
```

**Comment by piste-jp-ibm** [comment]
From the code, `stat()` call against `/dev/sg7` is failed because of a reason. Please list the device by `ltfs -o device_list` and specify a correct device.

https://github.com/LinearTapeFileSystem/ltfs/blob/618859c05444768e13963812918f22f5b062bd73/src/tape_drivers/linux/sg/sg_tape.c#L1242-L1262

**Comment by alexhabgood** [comment]
Thanks for the quick response. `ltfs -o device_list` fails to list the connected drives. There should be two:
Quantum LTO 6 Model B - /dev/sg6
Quantum LTO 6 Model C - /dev/sg7


```
maceadmin@amacewrk05:/$ ltfs -o device_list
42b5 LTFS14000I LTFS starting, LTFS version 2.5.0.0 (Prelim), log level 2.
42b5 LTFS14058I LTFS Format Specification version 2.4.0.
42b5 LTFS14104I Launched by "ltfs -o device_list".
42b5 LTFS14105I This binary is built for Linux (x86_64).
42b5 LTFS14106I GCC version is 7.5.0.
42b5 LTFS17087I Kernel version: Linux version 5.4.0-70-generic (buildd@lgw01-amd64-039) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #78~18.04.1-Ubuntu SMP Sat Mar 20 14:10:07 UTC 2021 i386.
42b5 LTFS17089I Distribution: NAME="Ubuntu".
42b5 LTFS17089I Distribution: DISTRIB_ID=Ubuntu.
42b5 LTFS17085I Plugin: Loading "sg" tape backend.
Tape Device list:.
maceadmin@amacewrk05:/$ 
```

**Comment by piste-jp-ibm** [comment]
Could you provide the following 3 outputs?

1. `ls -l /dev/sg?`
2. `mount | grep ltfs`
3. `ps aux | grep ltfs`

**Comment by alexhabgood** [comment]
Here you go:

```
maceadmin@amacewrk05:/media$ ls -l /dev/sg?
crw-rw----  1 root disk  21, 0 Sep 14 10:44 /dev/sg0
crw-rw----+ 1 root cdrom 21, 1 Sep 14 10:44 /dev/sg1
crw-rw----  1 root disk  21, 2 Sep 14 10:44 /dev/sg2
crw-rw----  1 root disk  21, 3 Sep 14 10:44 /dev/sg3
crw-rw----  1 root disk  21, 4 Sep 14 10:44 /dev/sg4
crw-rw----  1 root disk  21, 5 Sep 14 10:44 /dev/sg5
crw-rw----  1 root tape  21, 6 Sep 14 11:41 /dev/sg6
crw-rw----  1 root tape  21, 7 Sep 14 11:41 /dev/sg7
maceadmin@amacewrk05:/media$ mount | grep ltfs
maceadmin@amacewrk05:/media$ ps aux | grep ltfs
maceadm+ 17845  0.0  0.0  14432  1148 pts/0    S+   12:35   0:00 grep --color=auto ltfs
maceadmin@amacewrk05:/media$ ls -l /dev/sg?
crw-rw----  1 root disk  21, 0 Sep 14 10:44 /dev/sg0
crw-rw----+ 1 root cdrom 21, 1 Sep 14 10:44 /dev/sg1
crw-rw----  1 root disk  21, 2 Sep 14 10:44 /dev/sg2
crw-rw----  1 root disk  21, 3 Sep 14 10:44 /dev/sg3
crw-rw----  1 root disk  21, 4 Sep 14 10:44 /dev/sg4
crw-rw----  1 root disk  21, 5 Sep 14 10:44 /dev/sg5
crw-rw----  1 root tape  21, 6 Sep 14 11:41 /dev/sg6
crw-rw----  1 root tape  21, 7 Sep 14 11:41 /dev/sg7
maceadmin@amacewrk05:/media$ mount | grep ltfs
maceadmin@amacewrk05:/media$ ps aux | grep ltfs
maceadm+ 17853  0.0  0.0  14432  1112 pts/0    S+   12:37   0:00 grep --color=auto ltfs
maceadmin@amacewrk05:/media$ 

```

**Comment by piste-jp-ibm** [comment]
Hmm, it looks good...

It is a basic thing but is user `maceadmin` a member of group `tape`? You can see it by `id` command. So not, please try to list the device by `sudo` command like `sudo ltfs -o device_list`

**Comment by alexhabgood** [comment]
Well this is a bit strange. Just before you posted the last message I tested it again, and it's decided to start working. Not sure what happened there?!?

I also added `maceadmin` to the `tape` group. And now `ltfs -o device_list` is returning info:

maceadmin@amacewrk05:~$ ltfs -o device_list
1290 LTFS14000I LTFS starting, LTFS version 2.5.0.0 (Prelim), log level 2.
1290 LTFS14058I LTFS Format Specification version 2.4.0.
1290 LTFS14104I Launched by "ltfs -o device_list".
1290 LTFS14105I This binary is built for Linux (x86_64).
1290 LTFS14106I GCC version is 7.5.0.
1290 LTFS17087I Kernel version: Linux version 5.4.0-70-generic (buildd@lgw01-amd64-039) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #78~18.04.1-Ubuntu SMP Sat Mar 20 14:10:07 UTC 2021 i386.
1290 LTFS17089I Distribution: NAME="Ubuntu".
1290 LTFS17089I Distribution: DISTRIB_ID=Ubuntu.
1290 LTFS17085I Plugin: Loading "sg" tape backend.
Tape Device list:.
Device Name = /dev/sg1 (0.0.1.0), Vendor ID = QUANTUM , Product ID = ULTRIUM-HH6     , Serial Number = 91WT082810, Product Name =[ULTRIUM-HH6].

So that's another mystery solved. In any case, seems to be working now. So I'll close this issue, thanks for your help!
